clang/test/SemaSYCL/lb_sm_90.cpp

```diff
@@ -1,5 +1,7 @@
 // RUN: %clang_cc1 -internal-isystem %S/Inputs %s -triple nvptx64-nvidia-cuda -target-cpu sm_90 -fsycl-is-device -fsyntax-only -Wno-c++23-extensions -verify -S -o %t
 
+// REQUIRES: nvptx64-nvidia-cuda
```

Comment:
Can you explain why this is needed? I would have expected the triple to be enough for FE diagnostics.

Reply:
Thanks @elizabethandrews for reviews.

Test failed with error like:
```
********************
FAIL: Clang :: SemaSYCL/lb_sm_90.cpp (18707 of 19618)
******************** TEST 'Clang :: SemaSYCL/lb_sm_90.cpp' FAILED ********************
Exit Code: 1

Command Output (stderr):
--
RUN: at line 1: /bin/clang -cc1 -internal-isystem /lib/clang/19/include -nostdsysteminc -internal-isystem /llvm/clang/test/SemaSYCL/Inputs /llvm/clang/test/SemaSYCL/lb_sm_90.cpp -triple nvptx64-nvidia-cuda -target-cpu sm_90 -fsycl-is-device -fsyntax-only -Wno-c++23-extensions -verify -S -o /tools/clang/test/SemaSYCL/Output/lb_sm_90.cpp.tmp
+ /bin/clang -cc1 -internal-isystem /lib/clang/19/include -nostdsysteminc -internal-isystem /llvm/clang/test/SemaSYCL/Inputs /llvm/clang/test/SemaSYCL/lb_sm_90.cpp -triple nvptx64-nvidia-cuda -target-cpu sm_90 -fsycl-is-device -fsyntax-only -Wno-c++23-extensions -verify -S -o /tools/clang/test/SemaSYCL/Output/lb_sm_90.cpp.tmp
error: 'expected-error' diagnostics seen but not expected:
  (frontend): unable to create target: 'No available targets are compatible with triple "nvptx64-nvidia-cuda"'
1 error generated.

--
```
The CI runs by default with --cuda --hip configuration, so the test needs to be gated when not configured with --cuda or --hip. Lit test SemaSYCL/lb_sm_90.cpp that requires nvptx target to be built in order to properly run. This patch adds this requirement to the test.

Reply:
Thanks for the explanation.   I will approve, but would like @elizabethandrews to confirm she is okay with this.

Reply:
Thanks @premanandrao. 

>>but would like @elizabethandrews to confirm she is okay with this.

Sure. Sounds good to me.

Reply:
I assume the issue is with option `-target-cpu sm_90`  ? Without this option does the test pass?